tests/integration/state.go

```diff
@@ -50,11 +51,41 @@ type state struct {
 	// These channels will recieve a function which asserts results of any subscription requests.
 	subscriptionResultsChans []chan func()
 
-	// These synchronisation channels allow async actions to track their completion.
-	syncChans []chan struct{}
+	// nodeMergeCompleteSubs is a list of all merge complete event subscriptions
+	nodeMergeCompleteSubs []*event.Subscription
 
-	// eventSubs is a list of all event subscriptions
-	eventSubs []*event.Subscription
+	// nodeUpdateSubs is a list of all update event subscriptions
+	nodeUpdateSubs []*event.Subscription
```

Comment:
suggestion: Can we wrap these up in a dedicated type?  They are consuming a disproportionately large amount of attention on the state struct compared to their visibility needs.

Reply:
Done